zk_toolbox/crates/zk_inception/src/commands/prover/args/init.rs

```diff
@@ -0,0 +1,176 @@
+    pub credentials_file: String,
+}
+
+#[derive(Debug, Clone)]
+pub enum ProverInitArgsFinal {
+    FileBacked(ProofStorageFileBacked),
+    GCS(ProofStorageGCS),
+    GCSCreateBucket(ProofStorageGCSCreateBucket),
+}
+
+impl ProverInitArgs {
+    pub(crate) fn fill_values_with_prompt(&self) -> ProverInitArgsFinal {
+        if self.proof_store_dir.is_some() {
+            return self.handle_file_backed_config();
+        }
+
+        if self.partial_gcs_config_provided() {
+            return self.ask_gcs_config();
+        }
+
+        if self.partial_create_gcs_bucket_config_provided() {
+            return self.handle_create_gcs_bucket();
+        }
+
+        let proof_store_config =
+            PromptSelect::new(MSG_PROOF_STORE_CONFIG_PROMPT, ProofStoreConfig::iter()).ask();
+
+        match proof_store_config {
+            ProofStoreConfig::Local => self.handle_file_backed_config(),
+            ProofStoreConfig::GCS => self.handle_gcs_config(),
+        }
+    }
+
+    fn partial_create_gcs_bucket_config_provided(&self) -> bool {
+        self.create_gcs_bucket_config.bucket_name.is_some()
+            || self.create_gcs_bucket_config.location.is_some()
+            || self.create_gcs_bucket_config.project_id.is_some()
+    }
+
+    fn partial_gcs_config_provided(&self) -> bool {
+        self.proof_store_gcs_config.bucket_base_url.is_some()
+            || self.proof_store_gcs_config.credentials_file.is_some()
+    }
+
+    fn handle_file_backed_config(&self) -> ProverInitArgsFinal {
+        let proof_store_dir = self
+            .proof_store_dir
+            .clone()
+            .unwrap_or_else(|| Prompt::new(MSG_PROOF_STORE_DIR_PROMPT).ask());
+
+        ProverInitArgsFinal::FileBacked(ProofStorageFileBacked { proof_store_dir })
+    }
+
+    fn handle_gcs_config(&self) -> ProverInitArgsFinal {
+        if !self.partial_gcs_config_provided() {
+            if PromptConfirm::new(MSG_CREATE_GCS_BUCKET_PROMPT).ask() {
+                return self.handle_create_gcs_bucket();
+            }
+        }
+
+        self.ask_gcs_config()
+    }
+
+    fn handle_create_gcs_bucket(&self) -> ProverInitArgsFinal {
+        let project_id = self
+            .create_gcs_bucket_config
+            .project_id
+            .clone()
+            .unwrap_or_else(|| Prompt::new(MSG_CREATE_GCS_BUCKET_PROJECT_ID_PROMPT).ask());
+        let bucket_name = self
+            .create_gcs_bucket_config
+            .bucket_name
+            .clone()
+            .unwrap_or_else(|| Prompt::new(MSG_CREATE_GCS_BUCKET_NAME_PROMTP).ask());
+        let location = self
+            .create_gcs_bucket_config
+            .location
+            .clone()
+            .unwrap_or_else(|| Prompt::new(MSG_CREATE_GCS_BUCKET_LOCATION_PROMPT).ask());
+        let credentials_file = self
+            .clone()
+            .proof_store_gcs_config
+            .credentials_file
+            .unwrap_or_else(|| Prompt::new(MSG_PROOF_STORE_GCS_CREDENTIALS_FILE_PROMPT).ask());
```

Comment:
Probably we can use default from here 
https://cloud.google.com/docs/authentication/application-default-credentials